src/rules/consentric-integration.js

```diff
@@ -0,0 +1,183 @@
+
+    // Generates On Demand Consentric User Token for the given User using the API Access Token
+    const generateConsentricUserAccessToken = async ({ userRef, apiAccessToken }) => {
+        try {
+            console.log(`Attempting to generate access token API for ${userRef}`);
+
+            const { data: { token, expiryDate: exp } } = await consentricApi
+                .post('/v1/access-tokens/tokens',
+                    {
+                        applicationId: CONSENTRIC_APPLICATION_ID,
+                        externalRef: userRef,
+                        expiryDate: moment().add(3, 'months').toISOString()
+                    },
+                    {
+                        headers: {
+                            Authorization: 'Bearer ' + apiAccessToken
+                        }
+                    }
+                );
+
+            return {
+                token,
+                exp
+            };
+
+        } catch (err) {
+            console.error(err);
+            throw err;
+        }
+    };
+
+    const loadConsentricUserAccessToken = async ({ user }) => {
+        try {
+            const metadataUserToken = getConsentricUserTokenFromMetadata(user);            
+            if ((metadataUserToken) && moment(metadataUserToken.exp).isAfter(moment())) return metadataUserToken;
+        
+            const { jwt: apiAccessToken } = await getConsentricApiAccessToken();            
+            const apiCredentials = {
+                userRef: user.user_id,
+                apiAccessToken,
+            };
+
+            // Create Citizen with Auth0 UserId
+            await createCitizen(apiCredentials);
+
+            // Generate an On Demand Access Token for the created citizen
+            const generatedToken = await generateConsentricUserAccessToken(apiCredentials);
+
+            // Persist the app_metadata update            
+            await auth0.users.updateAppMetadata(user.user_id, { ...user.app_metadata, consentric: generatedToken });
+
+            return generatedToken;
+        } catch (err) {
+            console.error(`Issue Loading Consentric User Access Token for user ${user.user_id} - ${err}`);
+            throw err;
+        }
+    };
+
+    const initConsentricFlow = async () => {
+        try {
+            const { token } = await loadConsentricUserAccessToken({ user });
+            const urlConnector = CONSENTRIC_REDIRECT_URL.includes('?') ? '&' : '?';
+            const redirectUrl = CONSENTRIC_REDIRECT_URL + urlConnector + 'token=' + token;
+    
+            context.redirect = {
+                url: redirectUrl
+            };
+    
+            return callback(null, user, context);
+        } catch (err) {
+            console.error(`CONSENTRIC RULE ABORTED: ${err}`);
+            return callback(null, user, context);
+        }
+    };
+
+    if (ruleUtils.canRedirect) {
+        return initConsentricFlow();
+    } else {
+        // Run after Redirect or Silent Auth
```

Comment:
I'm not sure what kind of thing you mean here?

Reply:
As in ... applications will be requesting logins and expecting a response from Auth0. The response right now would not include anything from Consentric at all. Should there be some kind of indicator or ID that the application can use to determine status or anything? Answer could be "no" but just wanted to point that out as a possibility. 

If there should be something included, best place for that is in the ID token - `context.idToken` - in a [namespaced claim](https://auth0.com/docs/tokens/create-namespaced-custom-claims).

Reply:
Ah I see. We've tried to go for a light touch. Hopefully interfering with as little as possible in Customer's existing auth0 flow. Internally we've actually published a second rule to make the consentricUserToken available in the user object.

We've set this up as an optional separate rule in case customers wish to use the consentric token to surface additional widgets in their applications.

```
function (user, context, callback) {
    if(user.app_metadata && user.app_metadata.consentric) {
      context.idToken['https://consentric.io/consentric_user_token'] = user.app_metadata.consentric.token;
      context.idToken['https://consentric.io/consentric_user_token_expiry'] = user.app_metadata.consentric.exp;    
      context.accessToken['https://consentric.io/consentric_user_token'] = user.app_metadata.consentric.token;
      context.accessToken['https://consentric.io/consentric_user_token_expiry'] = user.app_metadata.consentric.exp;    
    }
    
    callback(null, user, context);
  }
```
We opted not to pollute the core one if the Customers weren't interested in utilising the consentric functionality further.

Reply:
OK, that makes sense. So maybe just publish this secondary Rule with the guide?